instrumentation/jaxws/jaxws-common/library/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxws/common/JaxWsServerSpanNaming.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jaxws.common;
+
+import static io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming.Source.CONTROLLER;
+import static io.opentelemetry.javaagent.instrumentation.jaxws.common.JaxWsSingletons.spanNameExtractor;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
+import java.util.function.Supplier;
+
+public class JaxWsServerSpanNaming {
+
+  public static void updateServerSpanName(Context parentContext, JaxWsRequest request) {
+    ServerSpanNaming.updateServerSpanName(
+        parentContext, CONTROLLER, getServerSpanNameSupplier(parentContext, request));
+  }
+
+  private static Supplier<String> getServerSpanNameSupplier(Context context, JaxWsRequest request) {
+    return () -> getServerSpanName(context, request);
+  }
+
+  private static String getServerSpanName(Context context, JaxWsRequest request) {
+    return spanNameExtractor().extract(request);
```

Comment:
maybe store spanName in request? using span name extractor here feels odd, but wonder what others think..

Reply:
Moved to request though it will still read `this.spanName = spanNameExtractor().extract(this);`